packages/functional-tests/pages/relier.ts

```diff
@@ -51,26 +51,32 @@ export class RelierPage extends BaseLayout {
   }
 
   async clickSubscribe() {
-    const waitForNavigation = this.page.waitForNavigation();
     await this.page
       .getByRole('link', { name: 'Subscribe to Pro (USD)' })
       .click();
-    return waitForNavigation;
+    // Depending on if the user is signed in or not, match /products/ or /checkout/.
+    // We need to add a `waitUntil` option here because the page gets redirected from
+    // content-server to payments-server, and the URL changes before the page is fully loaded.
+    return this.page.waitForURL(/\/(checkout|products)\//, {
+      waitUntil: 'networkidle',
+    });
```

Comment:
Wouldn't this pattern match both the content server page URL and the payments server page URL for existing FxA users? I.e. we'd still only wait for the first navigation[1] and not the second?

[1]: sample content server URL from stage (note presence of **/products/**):

https://accounts.stage.mozaws.net/subscriptions/products/prod_FiJ42WCzZNRSbS?plan=price_1J0Y1iKb9q6OnNsLXwdOFgDr...

[2]: sample payments server page URL for existing FxA users (note presence of **/products/**):

https://payments-stage.fxa.nonprod.cloudops.mozgcp.net/products/prod_FiJ42WCzZNRSbS?plan=price_1J0Y1iKb9q6OnNsLXwdOFgDr...